### PR TITLE
Add MAE for validation and update the interface of movielens dataset module

### PR DESCRIPTION
**Pull request by yunhuiguo**
## What changes were proposed in this pull request?
1. Add MAE for validation.
2. Update the interface of movielens dataset module.
3. Correct a typo in createTop5Accuracy().

## How was this patch tested?
unittest

### CODE REVIEW

**Comment by yunhuiguo** [comment]
pr validation pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala [hunk 1]
```diff
@@ -270,3 +271,23 @@ class Loss[@specialized(Float, Double)T: ClassTag](
...
   override def format(): String = "Loss"
 }
+
+/**
+ * This evaluation method is calculate mean absolute error of output with respect to target
+ *
+ */
+class MAE[@specialized(Float, Double)T: ClassTag]()
+(implicit ev: TensorNumeric[T]) extends ValidationMethod[T] {
+  var criterion = AbsCriterion[T]()
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala]: why not val? make it private?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala [hunk 2]
```diff
@@ -278,7 +278,7 @@ class Loss[@specialized(Float, Double)T: ClassTag](
  */
 class MAE[@specialized(Float, Double)T: ClassTag]()
 (implicit ev: TensorNumeric[T]) extends ValidationMethod[T] {
-  var criterion = AbsCriterion[T]()
+  val criterion = AbsCriterion[T]()
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala]: should it be private?

Comment (yunhuiguo) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala]: updated and pass pr validation.

**Comment by yiheng** [comment]
LGTM

**Comment by yiheng** [comment]
There's rebase conflict

**Comment by yunhuiguo** [comment]
Revolved conflicts and passed pr validation.

**Pull request merged**